sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.ts

```diff
@@ -464,4 +468,71 @@ export class StudentAESTController extends BaseController {
       );
     }
   }
+
+  /**
+   * Get possible matches for the student from the legacy system.
+   * @param studentId student ID to retrieve the data.
+   * @returns student legacy profile details.
+   */
+  @Roles(Role.AESTStudentLinkLegacyProfile)
+  @Get(":studentId/legacy-match")
+  @ApiNotFoundResponse({ description: "Student not found." })
+  @ApiUnprocessableEntityResponse({
+    description: "Student already has a legacy profile associated.",
+  })
```

Comment:
Do we require this validation in the GET API. I believe it is not an API consideration for the GET API.

Reply:
The method right now should not be invoked for any other reason. If someday it will be reused, it can be easily adapted, assuming this day will come before we remove the SFAS-related code.
Please let me know if it is a blocker.

Reply:
It is not a blocker.